packages/nimbus/src/components/money-input/money-input.tsx

```diff
@@ -122,28 +122,23 @@ export const MoneyInputComponent = (props: MoneyInputProps) => {
     placeholder = "0.00",
     autoFocus,
     size,
-    "aria-label": ariaLabel,
-    "aria-labelledby": ariaLabelledBy,
     ...restProps
   } = props;
 
   // Generate IDs
-  const generatedId = useId();
-  const id = providedId || generatedId;
+  const defaultGroupId = useId();
+  const groupId = id ?? defaultGroupId;
 
   // Get locale for formatting
   const { locale } = useLocale();
   const intl = useIntl();
 
-  // State management
-  const [currencyHasFocus, setCurrencyHasFocus] = useState(false);
-  const [amountHasFocus, setAmountHasFocus] = useState(false);
-  const hasFocus = currencyHasFocus || amountHasFocus;
```

Comment:
don't need focus state management here with `focusWithin`, and using the react-aria-components `Input` as the base component for the number input input slot